agent/consul/leader_metrics.go

```diff
@@ -43,6 +51,28 @@ func rootCAExpiryMonitor(s *Server) certExpirationMonitor {
 	}
 }
 
+func intermediateCAExpiryMonitor(s *Server) certExpirationMonitor {
+	return certExpirationMonitor{
+		Key: metricsKeyMeshCAExpiry["intermediate-cert"],
+		Labels: []metrics.Label{
+			{Name: "datacenter", Value: s.config.Datacenter},
+		},
+		Logger: s.logger.Named(logging.Connect),
+		Query: func() (time.Duration, error) {
+			provider, _ := s.caManager.getCAProvider()
+			certPem, err := provider.ActiveIntermediate()
```

Comment:
I need to look into this more next week. The current implementation of the active-root-ca metric uses the state store to fetch the certificate, where as this is using the provider. I'm not sure if those could ever return different answers, or what the implications are of using one or the other. 

I'd like to understand that, and possibly have both metrics use the same source of data.

Despite the name of the method, `CARootActive` actually returns a `struct.CARoot` which also has a slice of Intermediate certs on it. So we could get both sets of data from the same source, I'm not sure which is more correct.

Reply:
Yes I have the same questions. I was also tempted to change the root metric to use the same data source for the root, also we need to handle the case were we use a Vault/AWS provider. In that case the root CA is not ours and we use an intermediate. My understanding is that the current root metric won't be accurate in that case.

Reply:
I changed to use the same way to access data